src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/JavaScriptTestHelper.cs

```diff
@@ -1000,14 +1000,16 @@ public static async Task InitializeAsync()
             if (_module == null)
             {
                 _module = await JSHost.ImportAsync("JavaScriptTestHelper", "../JavaScriptTestHelper.mjs"); ;
-                await Setup(); ;
+                await Setup();
             }
 
-            var p = echopromise_String("aaa");
-            await p;
-
-            // this gives browser chance to serve UI thread event loop before every test
-            await Task.Yield();
+#if FEATURE_WASM_THREADS
+            if (Environment.CurrentManagedThreadId == 1)
```

Comment:
if this is important it would be good to have an explanation of why we're doing this, i can imagine a few reasons